hazelcast-sql-core/src/test/java/com/hazelcast/sql/impl/expression/predicate/ComparisonPredicateIntegrationTest.java

```diff
@@ -219,13 +227,174 @@ public void testParameterParameter() {
     @Test
     public void testUnsupported() {
         // Column/column
-        checkUnsupportedColumnColumn(ExpressionTypes.LOCAL_DATE, ExpressionTypes.allExcept());
-        checkUnsupportedColumnColumn(ExpressionTypes.LOCAL_TIME, ExpressionTypes.allExcept());
-        checkUnsupportedColumnColumn(ExpressionTypes.LOCAL_DATE_TIME, ExpressionTypes.allExcept());
-        checkUnsupportedColumnColumn(ExpressionTypes.OFFSET_DATE_TIME, ExpressionTypes.allExcept());
+        checkUnsupportedColumnColumn(
```

Comment:
These tests are not aligned with the SQL standard expectations. E.g., `DATE` can't be cast to `TIME`. See the SQL standard [1], 5WD-02-Foundation-2003-09.pdf, paragraph 4.6.2, table 3.

[1] http://www.wiscorp.com/sql_2003_standard.zip